### PR TITLE
Add setInterruptBuffer to public API, update changelog and docs

### PR DESCRIPTION
**Pull request by hoodmane**
This minimally solves issue #676, though we may wish to wait to close it until we offer a more convenient webworker API.

### CODE REVIEW

File: docs/project/changelog.md [hunk 1]
```diff
@@ -42,6 +42,7 @@
   of Pyodide REPL in webpages (used in console.html) [#1125](https://github.com/iodide-project/pyodide/pull/1125) and [#1155](https://github.com/iodide-project/pyodide/pull/1155)
 - Flexible jsimports: it now possible to add custom Python "packages" backed by Javascript code, like the js package.
   The js package is now implemented using this system. [#1146](https://github.com/iodide-project/pyodide/pull/1146)
+- Added the `pyodide.setInterruptBuffer` API. This can be used to set a `SharedArrayBuffer` to be the keyboard interupt buffer. If Pyodide is running on a webworker, the main thread can signal to the webworker that it should raise a `KeyboardInterrupt` by writing to the interrupt buffer.
```
Comment (rth) [inline docs/project/changelog.md:45]: Please wrap it to be 80 char (`gq` in vim). That will make the diff shorter for future changes.

Also could you please add a link to https://github.com/iodide-project/pyodide/pull/1148 at the end.

File: docs/usage/js-api/pyodide.setInterruptBuffer.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+(js_api_pyodide_setInterruptBuffer)=
+# pyodide.setInterruptBuffer(interruptBuffer)
+This is a low level API for handling keyboard interrupts.
+Sets the pyodide interrupt buffer to be `interruptBuffer`. If thereafter one sets `interruptBuffer[0] = 2;` (2 stands for SIGINT) this will cause Pyodide to raise a `KeyboardInterupt`. The value of `interruptBuffer[0]` will regularly be set back to zero.
+This is intended for use when Pyodide is running on a webworker. In this case, one should make `interruptBuffer` a `SharedArrayBuffer` shared with the main thread. If the user requests a keyboard interrupt from the main thread, then the main thread can set `interruptBuffer[0] = 2;` and this will signal the webworker to raise a KeyboardInterupt exception.
```
Comment (rth) [inline docs/usage/js-api/pyodide.setInterruptBuffer.md:5]: Please also keep the maximum line width at 80 (or 88) char when possible.

**Review by rth — COMMENTED** [review]
Minor comments otherwise LGTM. Thanks!

**Comment by hoodmane** [comment]
@rth Okay I added the links to the relevant PRs and wrapped the lines at 80 chars using `gq80`. This changed some other stuff in the changelog but it looks like it'd been formatted with `gq80` not that long ago. I manually fixed a bunch of little issues though.

**Comment by rth** [comment]
Thanks!

**Pull request merged**